src/mds/MDSMap.h

```diff
@@ -350,6 +354,27 @@ class MDSMap {
     s = failed;
   }
 
+  // features
+  uint64_t get_up_features() {
+    if (!cached_up_features) {
+      bool first = true;
+      for (std::map<mds_rank_t, mds_gid_t>::const_iterator p = up.begin();
```

Comment:
Not necessary, but remember you can use the auto keyword for this nonsense now. :D